### ISSUE TITLE
DataTable: Sticky Header confused by margin

### ISSUE THREAD

**Issue by ami2009** [primefaces/primefaces#326]
Sticky Header of the Datatable Element gets confused if margins are used. 

attached is the example of your showcase application. 

Sticky Header shown twice.
![image](https://cloud.githubusercontent.com/assets/6146092/7766646/a3e9647a-006b-11e5-9080-3a22d658710a.png)

Sticky Header with wrong offset.
![image](https://cloud.githubusercontent.com/assets/6146092/7766660/ca9cb658-006b-11e5-9c97-ed9d618a7672.png)

**Comment by cagataycivici** [comment]
Margins used where? Sticky does not support responsive at the moment, if you mean losing position after resize of window.

**Comment by ami2009** [comment]
The primefaces showcase application as an example: 

The element CONTENTSIDE has a left margin of 300px if the menu is visible. If you now click on datatable sticky header example everything works as expected. If the browser window is now resized to a smaller size so the responsive menu collapses, the sticky header still has a left position of 330 (300px I guess was the margin of the old layout element).

I guess the sticky header is simply not recalculated at the right time.

**Comment by cagataycivici** [comment]
It is calculated at load time so not responsive. We'll check this some time later as 5.3 focuses on responsiveness of the components. Thanks for the feedback.

I believe listening to window resize event and recalculating left position can fix it. Feel free to create a pull request.

**Comment by ami2009** [comment]
...it also happens when you use the datatable sticky header inside a ScrollPanel. I wanted to use it within a layoutunit position south which will not work at all. The sticky haeder is all over the place.

You are welcome

**Comment by cagataycivici** [comment]
Sorry, that is also not supported right now. Feel free to provide an enhancement path.

**Comment by beevelop** [comment]
I know this issue is quite old, but I still experienced the same inconsistencies with Sticky header's positioning with PrimeFaces 5.3
Fortunately I stumbled upon the jQuery lib [floatThead](https://github.com/mkoryak/floatThead) and successfully got a fully functioning version running. As floatThead is quite similar to PrimeFace's own approach to clone the thead and place it outside the original table, I only needed to mirror / hook into the "thead interactions": especially resizing, reordering and filtering columns.

Are there any plans to incorporate such libraries instead of trying to build such functionality from scratch?

Edit: This issue might be related to #834
